src/ts/ConvivaAnalytics.ts

```diff
@@ -130,51 +135,78 @@ export class ConvivaAnalytics {
-    let assetTitle = source.title;
-    let assetName;
+    let source = this.player.getConfig().source;
+    if (source) {
+      this.contentMetadata.assetName = this.getAssetName(source);
+      this.contentMetadata.streamUrl = this.getUrlFromSource(source);
 
-    if (assetId && assetTitle) {
-      assetName = `${assetId} ${assetTitle}`;
-    } else if (assetId && !assetTitle) {
-      assetName = assetId;
-    } else if (assetTitle && !assetId) {
-      assetName = assetTitle;
     } else {
-      assetName = 'Untitled (no source.title/source.contentId set)';
+      this.contentMetadata.assetName = 'unknown';
+      this.contentMetadata.streamUrl = 'unknown';
     }
 
-    // Create a ContentMetadata object and supply relevant metadata for the requested content.
-    let contentMetadata = new Conviva.ContentMetadata();
-    contentMetadata.assetName = assetName;
-    contentMetadata.viewerId = source.viewerId || this.config.viewerId || null;
-    contentMetadata.applicationName = this.config.applicationName || 'Unknown (no config.applicationName set)';
-    contentMetadata.duration = this.player.getDuration(); // TODO how to handle HLS Chrome deferred duration detection?
-    contentMetadata.streamType = this.player.isLive() ? Conviva.ContentMetadata.StreamType.LIVE // TODO how to handle HLS deferred live detection?
-      : Conviva.ContentMetadata.StreamType.VOD;
-    contentMetadata.streamUrl = this.getUrlFromSource(source);
-    contentMetadata.custom = {
-      'playerType': this.player.getPlayerType(),
-      'streamType': this.player.getStreamType(),
-      'vrContentType': this.player.getVRStatus().contentType,
-    };
-
     // Create a Conviva monitoring session.
-    this.sessionKey = this.client.createSession(contentMetadata);
+    this.sessionKey = this.client.createSession(this.contentMetadata);
 
     if (!this.isValidSession()) {
       // Something went wrong. With stable system interfaces, this should never happen.
       this.logger.consoleLog('Something went wrong, could not obtain session key',
         Conviva.SystemSettings.LogLevel.ERROR);
     }
 
+    this.playerStateManager.setPlayerState(Conviva.PlayerStateManager.PlayerState.STOPPED);
     this.client.attachPlayer(this.sessionKey, this.playerStateManager);
     this.debugLog('startsession', this.sessionKey, event);
+  }
+
+  private updateSession = () => {
+    let source = this.player.getConfig().source;
+    // Create a ContentMetadata object and supply relevant metadata for the requested content.
+
+    if (this.contentMetadata.assetName !== this.getAssetName(source)) {
+      this.contentMetadata.assetName = this.getAssetName(source);
+    }
+
+    this.contentMetadata.viewerId = source.viewerId || this.config.viewerId || null;
+
+    this.contentMetadata.duration = 0; // TODO how to handle HLS Chrome deferred duration detection?
+    this.contentMetadata.streamType = Conviva.ContentMetadata.StreamType.UNKNOWN; // TODO how to handle HLS deferred
```

Comment:
You were working on improvements regarding reporting of `duration` and `streamType`. Do these `TODO`s here still make sense? Is the reporting still not working correctly? Or does it maybe only not work under certain conditions that could be documented somewhere?